dateutil/rrule.py

```diff
@@ -1563,8 +1564,14 @@ def _parse_rfc(self, s,
                     # RFC 5445 3.8.2.4: The VALUE parameter is optional, but
                     # may be found only once.
                     value_found = False
+                    TZID = None
                     valid_values = {"VALUE=DATE-TIME", "VALUE=DATE"}
                     for parm in parms:
+                        if parm.startswith("TZID="):
+                            tzkey = parm.split('TZID=')[-1]
+                            tzlookup = tzinfos.get if tzinfos else tz.gettz
```

Comment:
I think it's best to explicitly compare to `None`, since I would think that `tzinfos={}` should not be equivalent to `tzinfos=tz.gettz`.

Reply:
I guess I'm confused about _when_ you would want to use `tz.gettz`.  Wouldn't it be _any_ time `tzinfos` *wasn't* a dictionary with keys?

Reply:
@ryanpetrello I am fine with `tz.gettz` being the default if you don't specify anything, but `tzinfos` can also be a callable, like this:

```python
def my_tzinfos(tzid):
    if tzid in SOME_LOOKUP_TABLE:
        return tzid[SOME_LOOKUP_TABLE]
    else:
        try:
            return dateutil.tz.tzstr(tzid)
        except ValueError:
            return tz.tzutc()
```

I don't really know if that particular function is something someone would actually want to do, but the point is that `tzinfos` is basically a mapping between `tzid` and a `tzinfo` object, so any function that provides `tzinfo` as a function of `tzid` should work. As a convenience, you can also provide a dictionary directly, rather than passing `mytzinfos.get`.

Reply:
```python
if callable(tzinfos):
    tzlookup = tzinfos
elif isinstance(tzinfos, dict):
    tzlookup = tzinfos.get
else:
    tzlookup = tz.gettz
```

More like this?